buckets/buckets.go

```diff
@@ -212,7 +212,24 @@ func (b *Buckets) Save(ctx context.Context, dbID thread.ID, bucket *Bucket, opts
 	for _, opt := range opts {
 		opt(args)
 	}
-	return b.threads.Save(ctx, dbID, CollectionName, dbc.Instances{bucket}, db.WithTxnToken(args.Token))
+	ensureNoNulls(bucket)
+	err := b.threads.Save(ctx, dbID, CollectionName, dbc.Instances{bucket}, db.WithTxnToken(args.Token))
+	if err != nil {
+		if isInvalidSchemaErr(err) {
+			if err := b.updateCollection(ctx, dbID, opts...); err != nil {
+				return err
+			}
+			return b.Save(ctx, dbID, bucket, opts...)
+		}
```

Comment:
Maybe make `isInvalidSchemaErr` do the check `err != nil` and avoid some nesting?
Like:
```go
if isInvalidSchemaErr(err) {
   ...
}
if err != nil {
   return fmt.Errorf("saving bucket in thread: %s", err)
}
return nil

Reply:
Sure, can do